### PR TITLE
feat: ✨ add autocomplete pattern + helper

### LINKED ISSUE #269: feat: ✨ add autocomplete pattern + helper
## User Story

As a developer I want to be able to easily use an input with autosuggest functionality that fits to the Solid Design System.

## Acceptance criteria

- [x] Use the branch https://github.com/solid-design-system/solid/tree/feat/autocomplete as a base and optimize styles regarding to [Figma design](https://www.figma.com/file/VTztxQ5pWG7ARg8hCX6PfR/Solid-DS-%E2%80%93-Component-Library?type=design&node-id=401-2294&mode=design).
- [x] Describe why we decided to offer a helper for an existing library (https://tarekraafat.github.io/autoComplete.js/#/) instead of writing a component (a11y, keyboard handling, flexibility, bundle size etc. etc.)
- [x] helper function `packages/components/src/utilities/setup-autocomplete.ts`
  - [x] make it importable via ESM and usable via UMD
  - [x] describe its functionality, its defaults (injecting popup etc.) and which settings its offers
  - [x] describe its return value `config`
  - [x] describe how to utilize the script and config in combination with autocomplete.js
  - [x] point to the docs of autocomplete.js
- [x] Add additional docs and stories which show how to...
  - [x] describe the existing stories, what they can do and how they work
  - [x] use completely different elements like sd-option with icons or avatars etc. as an list element
  - [x] inject styles into the shadow-dom of `sd-input`
  - [ ] ~set custom height as requested in https://jira.apps.union-investment.de/browse/PB2B-3883~ -> #817 
  - [ ] ~You can set a max number of elements, the element resizes automatically regardign the space on the window and makes its content scrollable.~ -> #819
  - [ ] ~think of other examples, like fetching data async etc. (get inspiration from e. g. download-search or similar)~ -> #980 
- [x] write some tests to test `setup-autocomplete.ts`
- [ ] ~make the initial docs page that it doesn't show the first story and the args table (see e. g. https://github.com/synergy-design-system/synergy-design-system/blob/main/packages/docs/stories/templates/form.stories.tsx#L20) and do the same for `quote`~-> #905 
- [x] Double check that everything works when you use our package or CDN output (using in another project)

## Related tickets
- [ ] - ~https://jira.apps.union-investment.de/browse/PB2B-3883~ -> #817 

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by saemik94**
closes #269

### CODE REVIEW

**Comment by mariohamann** [comment]
I don't know what's happening with the input here...

**Comment by yoezlem** [comment]
Do I have to do something before using the pattern?
image

**Review by karlbaumhauer — APPROVED** [review]

File: packages/components/src/patterns/autocomplete/autocomplete.stories.ts [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+        Promise.all([customElements.whenDefined('sd-input'), customElements.whenDefined('sd-popup')]).then(() => {
+          /* Simple example */
+          const { config: simpleConfig } = setupAutocomplete('#async-example');
+
+          new autoComplete({
+            ...simpleConfig,
+            data: {
+              src: async query => {
+                await new Promise(resolve => setTimeout(resolve, 1000));
+                return data.src.map(item => ({ item }));
```
[the remark this answers is not in the document]
Comment (saemik94) [reply packages/components/src/patterns/autocomplete/autocomplete.stories.ts:513]: -> #980

**Comment by yoezlem** [comment]
> Do I have to do something before using the pattern? image

image

@mariohamann are you sure that it is working now? Still have issues.

**Comment by mariohamann** [comment]
You screenshotted the only one not working. :D Update on the way!

**Review by mariohamann — APPROVED** [review]
@saemik94 The teaser example could get some further love, or maybe should be switched in future (e. g. proper hover and focus states). For the moment I'm fine with it but we should iterate on that. Could you please setup a follow-up ticket for that? @coraliefeil Would that be fine for you if we would iterate on that later on?

**Review by yoezlem — APPROVED** [review]
LGTM

**Review by azraefendic — APPROVED** [review]

**Comment by coraliefeil** [comment]
review done - pls reassign when adjusted. THX

**Comment by coraliefeil** [comment]
Approved.

**Comment by rolbr** [comment]
Hello!

I'm currently snooping around the code / storybook to get a feel for what's on the horizon. One thing that caught my eye: the 'Highlight' example highlights only the **very first occurence** of the search term:
grafik

Our current implementation based on ui-autosuggest highlights **all occurences** of the search term:
grafik

Not sure whether this is a bug -- just wanted to point it out as it caught me slightly off guard 🙂

**Comment by karlbaumhauer** [comment]
@rolbr thx for the hint. I opened up a [feature request](https://github.com/solid-design-system/solid/issues/1001) and we will have a look into it.

FYI @mariohamann @yoezlem

**Pull request merged**